cmd/node-agent/main.go

```diff
@@ -0,0 +1,290 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-systemd/dbus"
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_3"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	myPodNameEnvName        = "MY_POD_NAME"
+	desiredConfigAnnotation = "node-agent.alpha.coreos.com/desired-config"
+	currentConfigAnnotation = "node-agent.alpha.coreos.com/current-config"
+	kubeletVersionKey       = "KUBELET_VERSION"
+	kubeletConfigKey        = "KUBELET_CONFIG"
+	kubeletFlagsKey         = "KUBELET_FLAGS"
+	kubeletEnvPath          = "/etc/kubernetes/kubelet.env"
+)
+
+var (
+	sysdConn       *dbus.Conn
+	client         clientset.Interface
+	configMapStore cache.Store
+
+	myPodName = os.Getenv(myPodNameEnvName)
+)
+
+func main() {
+	if myPodName == "" {
+		glog.Fatalf("%s env var not set.", myPodNameEnvName)
+	}
+	run()
+}
+
+func run() {
+	client = newAPIClient()
+	nodename, err := getNodeName()
+	if err != nil {
+		glog.Fatal(err)
+	}
+
+	glog.Infof("starting node agent, watching node: %s", nodename)
+
+	sysdConn, err = dbus.New()
+	if err != nil {
+		glog.Fatal(err)
+	}
+
+	nodeopts := api.ListOptions{
+		FieldSelector: fields.ParseSelectorOrDie(fmt.Sprintf("metadata.name=%s", nodename)),
+	}
+	_, nodeController := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(lo api.ListOptions) (runtime.Object, error) {
+				return client.Core().Nodes().List(nodeopts)
+			},
+			WatchFunc: func(lo api.ListOptions) (watch.Interface, error) {
+				return client.Core().Nodes().Watch(nodeopts)
+			},
+		},
+		&v1.Node{},
+		30*time.Second,
+		framework.ResourceEventHandlerFuncs{
+			UpdateFunc: nodeUpdateCallback,
+		},
+	)
+	var configmapopts api.ListOptions
+	var configMapController *framework.Controller
+	configMapStore, configMapController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(lo api.ListOptions) (runtime.Object, error) {
+				return client.Core().ConfigMaps(api.NamespaceSystem).List(configmapopts)
```

Comment:
Actually, you don't need the configmapopts at all if we're not changing them. Can just use `lo` as passed in.